core/src/main/java/org/springframework/security/converter/ObjectConverters.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.converter;
+
+import org.springframework.core.convert.converter.Converter;
+
+import java.net.URI;
+import java.net.URL;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * A collection of {@link Converter} implementations
+ * that are capable of converting from {@code Object} to another type.
+ *
+ * @author Joe Grandja
+ * @since 5.2
+ * @see Converter
+ */
+public final class ObjectConverters {
```

Comment:
Have we looked into using built in Spring types (i.e. ConversionService)? It seems we should probably try to use that over writing the conversion ourselves.

Reply:
Yes, I did originally check `Converter` impl's in framework when I first wrote `ClaimAccessor` but did not find anything that met the needs. I will check once again to make sure.

Reply:
@rwinch I checked again and the reason I did not reuse `Converter`(s) in framework is because most (if not all) `Converter` implementations in `org.springframework.core.convert.support` are package-private.

Also, I don't feel it's necessary to reuse `ConversionService` (or `GenericConversionService`) as it provides additional API's that I won't use so I'd like to keep things as simple as possible.

Reply:
I'm worry about adding these converters as we will likely need to support more complex things like timestamps that already have support for conversion. What's more is this is rather general, so it means that as we need more types of conversions we need to support them directly vs relying on ConversionService. Perhaps we could still leverage ConversionService, but the API itself accept a Converter that was adapted from ConversionService